src/rdkafka_request.c

```diff
@@ -1878,6 +1878,49 @@ void rd_kafka_ListGroupsRequest_versioned(rd_kafka_broker_t *rkb,
         rd_kafka_broker_buf_enq_replyq(rkb, rkbuf, replyq, resp_cb, opaque);
 }
 
+/**
+ * Send DescribeGroupsRequest supporting a versioned request.
+ */
+void rd_kafka_DescribeGroupsRequest_versioned(rd_kafka_broker_t *rkb,
+                                              const char **groups,
+                                              int group_cnt,
+                                              rd_kafka_replyq_t replyq,
+                                              rd_kafka_resp_cb_t *resp_cb,
+                                              void *opaque) {
+        rd_kafka_buf_t *rkbuf;
+        int i;
+        size_t size;
+        int16_t api_version;
+        size_t groups_arraycnt;
+
+        api_version = rd_kafka_broker_ApiVersion_supported(
+            rkb, RD_KAFKAP_DescribeGroups, 0, 4, NULL);
+
+        size = 4 /* rd_kafka_buf_write_arraycnt_pos */ +
+               1 /* IncludeAuthorizedOperations */ + 1 /* tags */;
+        for (i = 0; i < group_cnt; i++) {
```

Comment:
You don't really need to calculate the exact buffer size, it will grow as needed. I just typically make an assumption, like a group name is what, 30 characters? So
`..buf_new_flexver_request(...., 4 + (30 * group_cnt) + 1 + 1, ..)`

Reply:
done